background.js

```diff
@@ -1,49 +1,53 @@
-browser.pageAction.onClicked.addListener(async () => {
-  await browser.tabs.executeScript({
-    code: `document.getElementsByClassName("download")[0].click();`,
-  });
-
-  await new Promise(r => setTimeout(r, 500));
+const clickElement = async querySelector => {
+  await browser.tabs.executeScript({code: `
+    if (document.querySelector("${querySelector}"))
+      document.querySelector("${querySelector}").click();
+  `});
+}
+
+const waitForElement = async querySelector => {
+  let [isElement] = await browser.tabs.executeScript({code: `document.querySelector("${querySelector}") !== null;`});
+  let timeLeft = 1000;
+
+  while (!isElement && timeLeft > 0) {
+    await new Promise(r => setTimeout(r, 100));
+    [isElement] = await browser.tabs.executeScript({code: `document.querySelector("${querySelector}") !== null;`});
```

Comment:
Minor:

The code snippet that checks if the element is present is repetitive and prone to typos (the contents of the `code` parameter are not type-checked or validated in any way and will only cause errors during runtime).

You could ensure that the same `code` is used here and on line 9 by extracting it to a variable, e.g.:

```ts
const elementFoundScript = `document.querySelector(${querySelector}) !== null;`;
```